skbio/alignment/_alignment.py

```diff
@@ -1405,6 +1407,200 @@ def sequence_length(self):
+           amino acids colored by their hydrophobicity.
+
+           Load the AAIndex database [1]_ hydrophobicity index values [2]_:
+
+           >>> from collections import defaultdict
+           >>> import numpy as np
+           >>> hydrophobicity_idx = defaultdict(lambda: np.nan)
+           >>> hydrophobicity_idx.update({'A': 0.61, 'L': 1.53, 'R': 0.60,
+           ...                            'K': 1.15, 'N': 0.06, 'M': 1.18,
+           ...                            'D': 0.46, 'F': 2.02, 'C': 1.07,
+           ...                            'P': 1.95, 'Q': 0., 'S': 0.05,
+           ...                            'E': 0.47, 'T': 0.05, 'G': 0.07,
+           ...                            'W': 2.65, 'H': 0.61, 'Y': 1.88,
+           ...                            'I': 2.22, 'V': 1.32})
+
+           Define an alignment of protein sequences:
+
+           >>> from skbio import Alignment, Protein
+           >>> sequences = [Protein('VHLTPEEKSAVTALWGKVNVDEV--', id="seq1"),
+           ...              Protein('-GLSDGEWQLVLKVWGKVEGDLPGH', id="seq2"),
+           ...              Protein('-GLSDGEWQLVLKVWGKVETDITGH', id="seq3"),
+           ...              Protein('-VLSEGEWQLVLHVWAKVEADIAGH', id="seq4")]
+           >>> aln = Alignment(sequences)
+
+           Plot the alignment as a heatmap:
+
+           >>> fig = aln.heatmap(hydrophobicity_idx, fig_size=(15, 10),
+           ...                   legend_labels=['Hydrophilic', 'Medium',
+           ...                                  'Hydrophobic'],
+           ...                   cmap='Greens')
+
+        """
+        if sequence_order is not None:
+            sequence_order_set = set(sequence_order)
+            if len(sequence_order) != len(sequence_order_set):
+                raise ValueError(
+                    "Found duplicate sequence ID(s) in `sequence_order`.")
+            if sequence_order_set != set(self.ids()):
+                raise ValueError(
+                    "`sequence_order` must contain the same set of sequence "
+                    "IDs in the alignment.")
+        else:
+            sequence_order = self.ids()
+
+        mtx, stats = self._alignment_to_heatmap_matrix(value_map,
+                                                       sequence_order)
+
+        # heatmap code derived from the matplotlib gallery:
+        #     http://matplotlib.org/examples/pylab_examples/
+        #     colorbar_tick_labelling_demo.html
+        fig, ax = plt.subplots()
+        if fig_size is not None:
+            fig.set_size_inches(fig_size)
+
+        cax = ax.imshow(mtx, interpolation='nearest', cmap=cmap)
+
+        sequence_count = self.sequence_count()
+        y_ticks = [0] + list(range(3, sequence_count - 3, 3))
+        if sequence_count > 1:
+            # we have more than one sequence, so add the index of the last
+            # sequence
+            y_ticks += [sequence_count - 1]
```

Comment:
In regard to the above, this will exclude the last "step-able" sequence and then add only the last sequence. i.e.
`sequence_count = 15`
`y_ticks = [0, 3, 6, 9, 14]` not `y_ticks = [0, 3, 6, 9, 12, 14]` which may have been the intended behaviour.